packages/client/hmi-client/src/composables/project.ts

```diff
@@ -0,0 +1,236 @@
+		if (code && code.id) {
+			await addAsset(projectId, AssetType.Code, code.id);
+		}
+		return code;
+	}
+
+	async function uploadArtifactToProject(
+		progress: Ref<number>,
+		file: File,
+		userName: string,
+		projectId: IProject['id'],
+		description?: string
+	) {
+		const artifact = await ArtifactService.uploadArtifactToProject(
+			progress,
+			file,
+			userName,
+			description
+		);
+		if (artifact && artifact.id) {
+			await addAsset(projectId, AssetType.Artifacts, artifact.id);
+		}
+		return artifact;
+	}
+
+	async function createNewArtifactFromGithubFile(
+		repoOwnerAndName: string,
+		path: string,
+		userName: string,
+		projectId: IProject['id']
+	) {
+		const artifact = await ArtifactService.createNewArtifactFromGithubFile(
+			repoOwnerAndName,
+			path,
+			userName
+		);
+		if (artifact && artifact.id) {
+			await addAsset(projectId, AssetType.Artifacts, artifact.id);
+		}
+		return artifact;
+	}
+
+	async function createNewDatasetFromCSV(
+		progress: Ref<number>,
+		file: File,
+		userName: string,
+		projectId: IProject['id'],
+		description?: string
+	) {
+		const dataset = await DatasetService.createNewDatasetFromCSV(
+			progress,
+			file,
+			userName,
+			description
+		);
+		if (dataset && dataset.id) {
+			await addAsset(projectId, AssetType.Datasets, dataset.id);
+		}
+		return dataset;
+	}
+
+	async function createNewDatasetFromGithubFile(
+		repoOwnerAndName: string,
+		path: string,
+		userName: string,
+		projectId: IProject['id'],
+		url: string
+	) {
+		const dataset = await DatasetService.createNewDatasetFromGithubFile(
+			repoOwnerAndName,
+			path,
+			userName,
+			url
+		);
+		if (dataset && dataset.id) {
+			await addAsset(projectId, AssetType.Datasets, dataset.id);
+		}
+		return dataset;
+	}
+
+	async function addNewModelToProject(modelName: string, projectId: IProject['id']) {
+		const modelId = await ModelService.addNewModelToProject(modelName);
+		if (modelId) {
+			await addAsset(projectId, AssetType.Models, modelId);
+		}
+		return modelId;
+	}
+
+	async function saveDatasetFromSimulationResultToProject(
+		projectId: IProject['id'],
+		simulationId: string | undefined,
+		datasetName: string | null
+	) {
+		if (!simulationId) return null;
+		const dataset = await DatasetService.createDatasetFromSimulationResult(
+			projectId,
+			simulationId,
+			datasetName
+		);
+		if (dataset) {
+			await getActiveProject(projectId);
+		}
+		return dataset;
+	}
+
+	return {
+		activeProject: readonly(activeProject),
+		allProjects: readonly(allProjects),
+		getActiveProject,
+		getAllProjects,
+		addAsset,
+		deleteAsset,
+		create,
+		update,
+		remove,
+		getPublicationAssets,
+		getAssetIcon,
+		uploadCodeToProject,
+		uploadCodeToProjectFromGithub,
+		uploadArtifactToProject,
+		createNewArtifactFromGithubFile,
+		createNewDatasetFromCSV,
+		createNewDatasetFromGithubFile,
+		addNewModelToProject,
+		saveDatasetFromSimulationResultToProject
```

Comment:
Some of those methods have the word `project` and some don't.
Could you uniform this by just removing it everywhere. 
I expect that we are going to use them with the pattern `useProject.active` for example. So it will be easier to read.

Reply:
I would suggest keeping `project` rather than getting rid of it. If we renamed `activeProject` to `active` for example, we would have

```const { active } = useProject()```

which would be more confusing in my opinion, since the context of `active` would not immediately obvious.